### DISCUSSION TITLE
skrifa: outline transformation

### DISCUSSION THREAD

**Discussion by msiglreith** [googlefonts/fontations#891] [Q&A] [answered]
Hello!

I'm currently replacing `freetype` with `skrifa` for text hinting/scaling. Output so far looks really good! One feature I wasn't able to find a matching implementation for is applying transformations to the outline , in particular translations (http://freetype.org/freetype2/docs/reference/ft2-outline_processing.html#ft_outline_translate), before hinting to allow subpixel offsets. Is this supported already or are there plans for supporting it?

Thank you!

**Comment by drott** [comment, marked as answer]
Thanks for your interest. Usually transforms other than scaling, in other words: any transform that shears or rotates glyph contours usually leads to the hinting programs getting disabled. Do you have any particular font in which you know the hinting programs affect transformed (sheared, rotated) points? 

_Edit: s/translations/transforms/_

**Reply by msiglreith** [reply]
Thanks for the reponse!

> Do you have any particular font in which you know the hinting programs affect transformed (sheared, rotated) points?

No, I'm only interested in the translation case and not aware of anything like that. But after reading your response I feel like the current API is sufficient for my use case: My motivation was to realize light hinting with subpixel positioning by offsetting the outline and then apply the hinting
```rust
HintingMode::Smooth {
  lcd_subpixel: None,
  preserve_linear_metrics: true,
}
```
I feel like hinting and then subpixel positioning yields similar or better results which are sufficient (considering I'm not an expert on this topic).
